Compiling errors on macos Mojave

It seems this is because of missing libraries in recent macOS builds:

```
clang: warning: libstdc++ is deprecated; move to libc++ [-Wdeprecated]
ld: library not found for -lstdc++
```
Tried doing everything possible: install new non-Apple clang, llvm and libc++, adding them to the path and rbconfig.rb files... nothing. Have banged my head for 5 hours searching on StackOverflow for a solution.

Here's the detailed log:

```
Building native extensions.  This could take a while...
ERROR:  Error installing mini_racer:
        ERROR: Failed to build gem native extension.

    current directory: /Users/marius.ursache/.rvm/gems/ruby-2.3.1/gems/mini_racer-0.2.0/ext/mini_racer_extension
/Users/marius.ursache/.rvm/rubies/ruby-2.3.1/bin/ruby -r ./siteconf20180808-43820-8a1r2c.rb extconf.rb
checking for main() in -lpthread... yes
checking for main() in -lobjc... yes
creating Makefile

current directory: /Users/marius.ursache/.rvm/gems/ruby-2.3.1/gems/mini_racer-0.2.0/ext/mini_racer_extension
make "DESTDIR=" clean

current directory: /Users/marius.ursache/.rvm/gems/ruby-2.3.1/gems/mini_racer-0.2.0/ext/mini_racer_extension
make "DESTDIR="
compiling mini_racer_extension.cc
clang: warning: argument unused during compilation: '-rdynamic' [-Wunused-command-line-argument]
In file included from mini_racer_extension.cc:2:
In file included from /Users/marius.ursache/.rvm/rubies/ruby-2.3.1/include/ruby-2.3.0/ruby.h:33:
In file included from /Users/marius.ursache/.rvm/rubies/ruby-2.3.1/include/ruby-2.3.0/ruby/ruby.h:1988:
/Users/marius.ursache/.rvm/rubies/ruby-2.3.1/include/ruby-2.3.0/ruby/intern.h:47:19: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
void rb_mem_clear(register VALUE*, register long);
                  ^~~~~~~~~
/Users/marius.ursache/.rvm/rubies/ruby-2.3.1/include/ruby-2.3.0/ruby/intern.h:47:36: warning: 'register' storage class specifier is deprecated and incompatible with C++17 [-Wdeprecated-register]
void rb_mem_clear(register VALUE*, register long);
                                   ^~~~~~~~~
2 warnings generated.
linking shared-object mini_racer_extension.bundle
clang: warning: libstdc++ is deprecated; move to libc++ [-Wdeprecated]
ld: library not found for -lstdc++
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [mini_racer_extension.bundle] Error 1

make failed, exit code 2
```

--- a/ext/mini_racer_extension/extconf.rb
+++ b/ext/mini_racer_extension/extconf.rb
@@ -11,7 +11,7 @@ $CPPFLAGS += " -std=c++0x"
 $CPPFLAGS += " -fpermissive"
 $CPPFLAGS += " -Wno-reserved-user-defined-literal" if RUBY_PLATFORM =~ /darwin/
 
-$LDFLAGS.insert 0, " -stdlib=libstdc++ " if RUBY_PLATFORM =~ /darwin/
+$LDFLAGS.insert 0, $1.to_i < 18 ? " -stdlib=libstdc++ " : " -stdlib=libc++ " if RUBY_PLATFORM =~ /darwin(\d+)/
 
 if ENV['CXX']
   puts "SETTING CXX"

